connectors/sources/s3.py

```diff
@@ -81,27 +111,36 @@ async def _get_content(self, doc, region, timestamp=None, doit=None):
             dictionary: Document of file content
         """
         # Reuse the same for all files
-        if not doit:
+        if not (doit and self.enable_content_extraction):
             return
         filename = doc["filename"]
         bucket = doc["bucket"]
         if os.path.splitext(filename)[-1] not in SUPPORTED_FILETYPE:
             logger.debug(f"{filename} can't be extracted")
             return
+        if doc["size"] > DEFAULT_MAX_FILE_SIZE:
+            logger.warning(
+                f"File size for {filename} is larger than {DEFAULT_MAX_FILE_SIZE} bytes. Discarding the file content"
+            )
+            return
         logger.debug(f"Downloading {filename}")
         async with self.client(region_name=region) as s3:
             try:
                 resp = await s3.get_object(Bucket=bucket, Key=filename)
                 await asyncio.sleep(0)
-                data = ""
-                while True:
-                    chunk = await resp["Body"].read(ONE_MEGA)
-                    await asyncio.sleep(0)
-                    if not chunk:
-                        break
-                    data += chunk.decode("utf8", errors="ignore")
-                logger.debug(f"Downloaded {len(data)} for {filename}")
-                return {"_timestamp": timestamp, "text": data, "_id": doc["id"]}
+                file_content, chunk = BytesIO(), True
```

Comment:
I am going to propose a small improvement here to lower the CPU usage and prepare for scaling:

If on linux or macos write on disk in a temporary file instead of BytesIO (use aiofiles)
Use the system's `base64` tool to encode to base64, and read back that.

It will be MUCH faster, will prevent extra memory bloat.

code snippet for base64 encoding:

```
def to_b64(filename):
    cmd = f"base64 {filename} > {filename}.b64"
    logger.info(f"[{os.path.basename(filename)}] Running {cmd}")
    os.system(cmd)
    return f"{filename}.b64"
```

Reply:
Hi @tarekziade , we tried the suggested approach by creating a .b64 file 
However after read the contents from .b64 file, we assume that the content is not getting extracted properly by the pipeline.
We are getting below message in elastic document
[ "Processor 'attachment' in pipeline 'ent-search-generic-ingestion' failed with message 'Illegal base64 character a'" ].

We checked the base64 data with online converter and it decodes data properly.